### PR TITLE
AMIGAOS: Mutliple fixes

### PR DESCRIPTION
**Pull request by raziel-**
- Add FS directory creation
- Remove hardcoded newlib toolchain (there may be follow-up FR's)

### CODE REVIEW

File: backends/fs/amigaos/amigaos-fs.cpp [hunk 1]
```diff
@@ -384,7 +384,23 @@ Common::SeekableWriteStream *AmigaOSFilesystemNode::createWriteStream() {
 }
 
 bool AmigaOSFilesystemNode::createDirectory() {
-	warning("AmigaOSFilesystemNode::createDirectory() -> Not supported");
+	BPTR lock = IDOS->Lock(_sPath.c_str(), SHARED_LOCK);
+	if (lock) {
+		IDOS->UnLock(lock);
+		debug("AmigaOSFilesystemNode::createDirectory() -> Directory '%s' already exists!", _sPath.c_str());
+		return _bIsValid && _bIsDirectory; // TODO: Should we make sure it's really a directory?
```
Comment (bluegr) [inline backends/fs/amigaos/amigaos-fs.cpp:391]: I don't understand this comment. Is there a possibility that a node with _bIsDirectory = true is not a directory?

**Comment by sev-** [comment]
Could you please fix the commit log messages to have the proper "AMIGAOS: " prefixes?

**Comment by raziel-** [comment]
@sev- 

I have no idea how to do that from github web page. 
I can close and redo the PR?

**Comment by digitall** [comment]
@raziel- , @sev- : Should be fixed now. I pushed to fix the commit prefix and squash the trivial typo commit into the previous commit.

**Comment by raziel-** [comment]
@bluegr

Regarding the comment...this will be revised.

@capehill will (hopefully) join and help out, please stand by

**Comment by capehill** [comment]
@raziel-  and other devs

Is it so that before calling backend's createDirectory, FSNode actually checks the existence and type: https://github.com/scummvm/scummvm/blob/master/common/fs.cpp#L156

In that case, backend could be simplified to something like:

```
bool AmigaOSFilesystemNode::createDirectory() { 
    BPTR lock = IDOS->CreateDir(_sPath.c_str()); 
    if (lock) { 
        IDOS->UnLock(lock); 
        debug("AmigaOSFilesystemNode::createDirectory() -> Directory '%s' created", _sPath.c_str()); 
        _bIsValid = true; 
        _bIsDirectory = true; 
    } else { 
        warning("AmigaOSFilesystemNode::createDirectory() -> Failed to create '%s'", _sPath.c_str()); 
    }     
     
    return _bIsValid && _bIsDirectory; 
}
```

This is still untested.

**Comment by raziel-** [comment]
Would that be applicable?

Tested locally and working

**Comment by bluegr** [comment]
This looks good now, and is aligned with what other platforms do in the same function. Since the changes are working for you @raziel, they should be fine to merge.

Thanks for your work!

**Pull request merged**